File: app_python/app/mocks/redis.py
class MockRedis:
    def __init__(self):
        self.data = dict()

    async def get(self, key: str) -> str | None:
        return self.data.get(key)

    async def incr(self, key: str) -> None:
        self.data[key] = self.data.get(key, 0) + 1

File: app_typescript/src/mocks/redis.ts
import Database from "../interfaces/database";

class MockRedis implements Database {
  private data: Map<string, string | number>;
  
  constructor() {
    this.data = new Map();
  }

  async get(key: string) {
    return String(this.data.get(key) ?? 0);
  }

  async incr(key: string) {
    const value = Number(this.data.get(key) ?? 0) + 1;
    this.data.set(key, value);
    return value;
  }
}

export default MockRedis;
